### DISCUSSION TITLE
How to implement Firebase App Check debug token

### DISCUSSION THREAD

**Discussion by artrayd** [nuxt/nuxt#10179] [Questions (Nuxt 2)]
Hello developers,

Since Frebase 9 is available, I'm rewriting my code to move on it.

Thanks to Lupas's [article](https://lupas.medium.com/firebase-9-beta-nuxt-js-981cf3dac910), everything works well. Except for app check.

I'm doing it by Firebase official [documentation](https://firebase.google.com/docs/app-check/web/debug-provider#web-version-9):

```
import { initializeAppCheck, ReCaptchaV3Provider } from 'firebase/app-check'
if (process.browser) {
  let appCheckKey
  // key for debug and prod
  if (process.env.NODE_ENV !== 'production') {
    window.FIREBASE_APPCHECK_DEBUG_TOKEN = true
    appCheckKey = 'XXXXXXXX'
    console.log(appCheckKey)
  } else {
    appCheckKey = 'XXXXXXXX'
  }

  // init app check
  /* eslint-disable */
   const appCheck = initializeAppCheck(firebaseApp, {
    provider: new ReCaptchaV3Provider(appCheckKey),
    isTokenAutoRefreshEnabled: true,
  })
}

```

Inside console, I see:
 App Check debug token: XXXXXXXX. You will need to add it to your app's App Check settings in the Firebase console for it to work.
 
 It seems legit, but then another error starts to appear every few minutes:
 **Uncaught (in promise) cancelled**

And when I try to perform some firebase action, I get this error:
Uncaught (in promise) FirebaseError: AppCheck: Fetch server returned an HTTP error status. HTTP status: 403. (appCheck/fetch-status-error).

On the same production, the code works well. The problem is at the dev environment. 
Without app check, everything works well.

I found the solution five minutes later and didn't know how to delete the issue.
I should add the token I got to the Firebase debug tokens and not the opposite...
The option to generate tokens confused me. It works vice-versa.
**Updated code:**
```
if (process.browser) {
  const appCheckKey = 'XXX'
  // key for debug and prod
  if (process.env.NODE_ENV !== 'production') {
    window.FIREBASE_APPCHECK_DEBUG_TOKEN = true
  } 

  /* eslint-disable */
  const appCheck = initializeAppCheck(firebaseApp, {
    provider: new ReCaptchaV3Provider(appCheckKey),
    isTokenAutoRefreshEnabled: true,
  })
}
```

**Comment by PeterF2170** [comment]
Thank you, the process.browser fixed my issue for prod
